### ISSUE TITLE
remote ipam driver not applying

### ISSUE THREAD

**Issue by rmb938** [docker/docker#17277]
When creating a network with a remote driver and remote ipam driver it is always set to default.

docker version: ```Docker version 1.9.0-rc1, build 9291a0e```
docker into:
```
Containers: 13
Images: 9
Engine Version: 1.9.0-rc1
Storage Driver: devicemapper
 Pool Name: docker-253:0-33662528-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.595 GB
 Data Space Total: 107.4 GB
 Data Space Available: 10.91 GB
 Metadata Space Used: 2.691 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.14.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 2
Total Memory: 1.797 GiB
Name: ccompute1.rmb938.com
ID: IAQS:6E74:7NGG:5JOG:JXFM:26VD:IAQV:FZNU:E23J:QUAA:NI4O:DI3S
```

uname -a ```Linux ccompute1.rmb938.com 3.10.0-229.14.1.el7.x86_64 #1 SMP Tue Sep 15 15:05:51 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux```

How reproducible: Always

Steps to Reproduce:
1. Create a network with a remote driver and remote ipam driver
2. Inspect the network and see the ipam driver is set to default

Describe the results you received:
```
[root@ccompute1 ~]# docker network create --ipam-driver=carbon -d carbon net1
91bf32ca0e7f192696df31b087ee488b2e15592252d26c51f076e469207cb084
[root@ccompute1 ~]# docker network inspect net1
{
    "name": "net1",
    "id": "91bf32ca0e7f192696df31b087ee488b2e15592252d26c51f076e469207cb084",
    "scope": "local",
    "driver": "carbon",
    "ipam": {
        "driver": "default",
        "config": [
            {}
        ]
    },
    "containers": {}
}
```

Describe the results you expected:
```
[root@ccompute1 ~]# docker network create --ipam-driver=carbon1 -d carbon net1
91bf32ca0e7f192696df31b087ee488b2e15592252d26c51f076e469207cb084
[root@ccompute1 ~]# docker network inspect net1
{
    "name": "net1",
    "id": "91bf32ca0e7f192696df31b087ee488b2e15592252d26c51f076e469207cb084",
    "scope": "local",
    "driver": "carbon",
    "ipam": {
        "driver": "carbon",
        "config": [
            {}
        ]
    },
    "containers": {}
}
```

**Comment by cpuguy83** [comment]
Have you tried with master? I think this is fixed.

**Comment by mavenugo** [comment]
@rmb938 @cpuguy83 it is fixed in libnetwork and is being vendored-in as we speak.
once we merge the vendor in, it will be resolved.

**Comment by mavenugo** [comment]
@rmb938 @cpuguy83 scratch that.
yes. @cpuguy83 is right. it is solved by #17202 already.

**Comment by rmb938** [comment]
@cpuguy83 @mavenugo Yup just tried master seems to be working. Thanks.
